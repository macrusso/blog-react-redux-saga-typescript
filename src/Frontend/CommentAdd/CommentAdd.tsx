import React from 'react';
import { Formik } from 'formik';
import { ICommentPartial } from 'src/Entities';

interface ICommentAddProps {
  match: any;
  addComment: (comment: ICommentPartial) => void;
}

const CommentAdd: React.SFC<ICommentAddProps> = props => {
  const { addComment, match } = props;
  return (
    <div>
      <h1>Create a comment!</h1>
      <Formik
        initialValues={{ body: '' }}
        onSubmit={values => {
          addComment({ ...values, userId: 1, postId: 2 });
        }}
      >
        {({ values, handleChange, handleBlur, handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <input
              type="body"
              name="body"
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.body}
            />
            <button type="submit">Submit</button>
          </form>
        )}
      </Formik>
    </div>
  );
};

export default CommentAdd;
