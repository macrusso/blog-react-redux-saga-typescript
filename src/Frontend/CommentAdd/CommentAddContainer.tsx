import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { IAction, IStoreState } from '../../types';
import { ErrorBoundary } from '../Shared';
import {
  commentSelectors,
  ICommentPartial,
  commentActions,
  postSelectors,
} from '../../Entities';
import { CommentAdd } from '.';
import { appSelectors } from '../../App';

type ICommentAddContainerProps = IStateToProps & IDispatchToProps;

class CommentAddContainer extends Component<ICommentAddContainerProps> {
  public render() {
    const props = this.props;

    return (
      <ErrorBoundary>
        <CommentAdd {...props} />
      </ErrorBoundary>
    );
  }
}

interface IStateToProps {
  selectedPostId?: number;
  error?: string;
  userId: number;
}

interface IDispatchToProps {
  addComment: (comment: ICommentPartial) => void;
}

const mapStateToProps = (state: IStoreState) => ({
  selectedPostId: postSelectors.getSelectedPostId(state),
  error: commentSelectors.getError(state),
  userId: appSelectors.getUserId(state),
});

const mapDispatchToProps = (dispatch: Dispatch<IAction>) => ({
  addComment: (comment: ICommentPartial) => {
    dispatch(commentActions.addCommentRequest(comment));
  },
});

export default connect<IStateToProps, IDispatchToProps, any>(
  mapStateToProps,
  mapDispatchToProps
)(CommentAddContainer);
