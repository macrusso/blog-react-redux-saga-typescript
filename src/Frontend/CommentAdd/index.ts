import CommentAddContainer from "./CommentAddContainer";
import CommentAdd from "./CommentAdd";

export { CommentAddContainer, CommentAdd };
