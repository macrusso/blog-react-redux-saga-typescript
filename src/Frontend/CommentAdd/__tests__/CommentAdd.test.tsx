import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import CommentAdd from '../CommentAdd';

describe('CommentAdd', () => {
  test('Snapshot matches', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <CommentAdd addComment={jest.fn()} match={{}} />
      </BrowserRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
