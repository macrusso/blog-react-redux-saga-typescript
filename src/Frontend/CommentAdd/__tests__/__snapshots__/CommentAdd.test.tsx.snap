// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentAdd Snapshot matches 1`] = `ShallowWrapper {}`;
